crates/matrix-sdk/src/room/timeline/inner.rs

```diff
@@ -146,17 +146,26 @@ impl<P: ProfileProvider> TimelineInner<P> {
         .await
     }
 
+    /// Update the transaction ID by an event ID.
     pub(super) fn add_event_id(&self, txn_id: &TransactionId, event_id: OwnedEventId) {
         let mut lock = self.items.lock_mut();
         if let Some((idx, item)) = find_event_by_txn_id(&lock, txn_id) {
-            if let Some(existing_event_id) = &item.event_id {
-                error!(
-                    ?existing_event_id, new_event_id = ?event_id, ?txn_id,
-                    "Local echo already has an event ID"
+            // It's always a `TimelineKey::TransactionId`
```

Comment:
Maybe we should write something like this?

```rust
let TimelineKey::TransactionId { txn_id, event_id: txn_event_id } = &item.key else {
    unreachable!("find_event_by_txn_id can only find items with TimelineKey::TransactionId");
};
```

Reply:
:+1: